internal/buffer/save.go

```diff
@@ -24,7 +24,63 @@ import (
 // because hashing is too slow
 const LargeFileThreshold = 50000
 
-func (b *Buffer) writeFile(file io.Writer) (int, error) {
+type wrappedFile struct {
+	writeCloser io.WriteCloser
+	withSudo    bool
+	screenb     bool
+	cmd         *exec.Cmd
+	sigChan     chan os.Signal
+}
+
+func openFile(name string, withSudo bool) (wrappedFile, error) {
+	var err error
+	var writeCloser io.WriteCloser
+	var screenb bool
+	var cmd *exec.Cmd
+	var sigChan chan os.Signal
+
+	if withSudo {
+		cmd = exec.Command(config.GlobalSettings["sucmd"].(string), "dd", "bs=4k", "of="+name)
+		writeCloser, err = cmd.StdinPipe()
+		if err != nil {
+			return wrappedFile{}, err
+		}
+
+		sigChan = make(chan os.Signal, 1)
+		signal.Reset(os.Interrupt)
+		signal.Notify(sigChan, os.Interrupt)
+
+		screenb = screen.TempFini()
+		// need to start the process now, otherwise when we flush the file
+		// contents to its stdin it might hang because the kernel's pipe size
+		// is too small to handle the full file contents all at once
+		err = cmd.Start()
+		if err != nil {
+			screen.TempStart(screenb)
+
+			signal.Notify(util.Sigterm, os.Interrupt)
+			signal.Stop(sigChan)
+
+			return wrappedFile{}, err
+		}
+	} else {
+		writeCloser, err = os.OpenFile(name, os.O_WRONLY|os.O_CREATE, util.FileMode)
+		if err != nil {
+			return wrappedFile{}, err
+		}
+	}
+
+	return wrappedFile{writeCloser, withSudo, screenb, cmd, sigChan}, nil
+}
+
+func (wf wrappedFile) Write(b *Buffer) (int, error) {
+	enc, err := htmlindex.Get(b.Settings["encoding"].(string))
+	if err != nil {
+		return 0, err
```

Comment:
Because of this, `safeWrite()` will return `util.ErrOverwrite` while the file was not really overwritten, causing a false alarm and a spurious backup?